packages/plugin/package.json

```diff
@@ -0,0 +1,33 @@
+{
+  "name": "@deephaven/plugin",
+  "version": "0.46.0",
+  "description": "Deephaven JS Plugin Core",
+  "author": "Deephaven Data Labs LLC",
+  "license": "Apache-2.0",
+  "type": "module",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/deephaven/web-client-ui.git",
+    "directory": "packages/app-utils"
+  },
+  "source": "src/index.js",
+  "main": "dist/index.js",
+  "module": "dist/index.js",
+  "types": "dist/index.d.ts",
+  "engines": {
+    "node": ">=16"
+  },
+  "scripts": {
+    "build": "cross-env NODE_ENV=production run-p build:*",
+    "build:babel": "babel ./src --out-dir ./dist --extensions \".ts,.tsx,.js,.jsx\" --source-maps --root-mode upward"
+  },
+  "devDependencies": {},
+  "dependencies": {},
+  "peerDependencies": {},
```

Comment:
Looks like you're missing some dependencies here? This package definitely references `@deephaven/iris-grid`, `@deephaven/components`, `@deephaven/jsapi-types` from `TablePlugin`.
That's one thing kind of annoying about this - creating an `AuthPlugin`, you don't care about `@deephaven/iris-grid` at all, but you need to import this package which will have a dependency on that.

Reply:
Ya they're technically dev dependencies since they aren't needed to run, but the types are required for someone else to install the package and get type support. So that would make them regular dependencies

I can't think of a better way that doesn't require you to install iris-grid or components. Unless we want to duplicate the types across both or pull them into a separate, shared types package which doesn't really make sense for these types IMO.

If we want to consolidate the definitions, I think it's just something we have to accept as a trade-off. We could make packages for each plugin type, but I'm not really a big fan of adding more packages here just to save a few MB of disk space for a development environment